qiskit_ibm_runtime/api/rest/runtime_session.py

```diff
@@ -0,0 +1,53 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Runtime Session REST adapter."""
+
+from typing import Dict, Any
+from .base import RestAdapterBase
+from ..session import RetrySession
+
+
+class RuntimeSession(RestAdapterBase):
+    """Rest adapter for session related endpoints."""
+
+    URL_MAP = {
+        "self": "",
+        "close": "/close",
+    }
+
+    def __init__(self, session: RetrySession, session_id: str, url_prefix: str = "") -> None:
+        """Job constructor.
+
+        Args:
+            session: RetrySession to be used in the adapter.
+            session_id: Job ID of the first job in a runtime session.
+            url_prefix: Prefix to use in the URL.
+        """
+        super().__init__(session, "{}/sessions/{}".format(url_prefix, session_id))
+
+    def close(self) -> None:
+        """Close this session."""
+        url = self.get_url("close")
+        self.session.delete(url)
+
+    def details(self) -> Dict[str, Any]:
+        """Return the details of this session."""
+        try:
+            if "cloud" in self.session.base_url:
+                return self.session.get(self.get_url("self")).json()
+            else:
+                # TODO: remove this once "v2" is removed from the url path
+                return self.session.get(self.get_prefixed_url("/v2", "self")).json()
+        # return None if API is not supported
+        except:  # pylint: disable=bare-except
+            return None
```

Comment:
> Something to keep in mind with this, and I'm not sure how Qiskit works with this, but the GET /sessions/:id API is only being implemented in the ntc and sw-api-iqp (new job service) repos, not the ntc-iqp-channel fork. The ntc repo is deployed for cloud pay-go and IQP DE (Fraunhofer). The sw-api-iqp service is only going to be deployed in the US.

Copying these concerns from the original issue, I think this logic handles everything but if/when `/v2` is removed from the path, that would be a breaking change?

Reply:
Update, the `/v2` path will be removed, so we should wait for that